unittests/test_api.py

```diff
@@ -168,6 +168,27 @@ def test_get_engagement_found(self, mockGet, mockEnv):
         mockGet.assert_called_once_with(url, headers=self.header, params=payload, verify=True)
         response.raise_for_status.assert_called_once()
 
+    @patch('dd_import.environment.Environment')
+    @patch('requests.get')
+    @patch.dict('os.environ', {'DD_URL': 'https://example.com',
+                               'DD_API_KEY': 'api_key',
+                               'DD_ENGAGEMENT_NAME': 'engagement',
+                               'DD_ENGAGEMENT_VERSION': '1.0.1'})
+    def test_get_engagement_with_version_found(self, mockGet, mockEnv):
+        response = Mock(spec=Response)
+        response.status_code = 200
+        response.text = '{\"count\": 2, \"results\": [{\"id\": 2, \"name\": \"engagement_dev\", \"version\": \"null\"}, {\"id\": 3, \"name\": \"engagement\", \"version\": \"null\"},  {\"id\": 4, \"name\": \"engagement\", \"version\": \"1.0.1\"}]}'
+        mockGet.return_value = response
+
+        api = Api()
+        id = api.get_engagement(self.product_id)
```

Comment:
Why does this function return the first result?

Reply:
I guess you need to select someway. Taking the first is a reasonable option. In the end there is no clear way what to select if there are more than one matches. The only other reasonable way would be to fail the script if more than one entry is found. However this was already like this from the original author,